### ISSUE TITLE
github-annotation outputs unrelated run prints, not parseable JSON in 3.0.7

### ISSUE THREAD

**Issue by vholmer** [sqlfluff/sqlfluff#5933]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

In 3.0.6 if I run the following:

```sh
sqlfluff lint --processes=0 --format github-annotation --annotation-level failure file.sql
```

I'll get this output:

```sh
[]
```

In 3.0.7, however, I get this output:

```sh
10:44:32  Registered adapter: snowflake=1.8.2
10:44:32  [WARNING]: Configuration paths exist in your dbt_project.yml file which do not apply to any resources.
There are 1 unused configuration paths:
- models.elementary
[]
```

Naturally this is not a valid JSON file and not usable by the recommended yuzutech annotations action.

### Expected Behaviour

Expect valid JSON output parseable by yuzutech annotations-action.

### Observed Behaviour

Prints unrelated to expected annotations JSON.

### How to reproduce

Lint any file with --format github-annotation.

### Dialect

Snowflake

### Version

sqlfluff, version 3.0.7

### Configuration

[sqlfluff]
templater = dbt
dialect = snowflake
encoding = utf-8

[sqlfluff:templater:dbt]
project_dir = dbt
profiles_dir = dbt
profile = snowflake-profile
target = snowflake

[sqlfluff:templater:jinja]
apply_dbt_builtins = true

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
This should be fixed by #5907 which is now in main. Let me know if you see otherwise.

**Comment by alanmcruickshank** [comment]
I'm going to close this one - assuming it's now solved. Do re-open if it's not.
